src/librbd/exclusive_lock/PreReleaseRequest.cc

```diff
@@ -180,10 +182,12 @@ void PreReleaseRequest<I>::handle_invalidate_cache(int r) {
   if (r < 0 && r != -EBLACKLISTED && r != -EBUSY) {
     lderr(cct) << "failed to invalidate cache: " << cpp_strerror(r)
                << dendl;
-    m_image_ctx.io_work_queue->unblock_writes();
-    save_result(r);
-    finish();
-    return;
+    if(r != -EDQUOT) {
```

Comment:
Nit: same comment here. Just add `-EDQUOT` and `-ENOSPC` to the if clause exclusions.

Reply:
done, check this plesae